crates/matrix-sdk-indexeddb/src/state_store.rs

```diff
@@ -237,6 +238,70 @@ async fn backup(source: &IdbDatabase, meta: &IdbDatabase) -> Result<()> {
     Ok(())
 }
 
+fn serialize_event(store_cipher: Option<&StoreCipher>, event: &impl Serialize) -> Result<JsValue> {
+    Ok(match store_cipher {
+        Some(cipher) => JsValue::from_serde(&cipher.encrypt_value_typed(event)?)?,
+        None => JsValue::from_serde(event)?,
+    })
+}
+
+fn deserialize_event<T: DeserializeOwned>(
+    store_cipher: Option<&StoreCipher>,
+    event: JsValue,
+) -> Result<T> {
+    match store_cipher {
+        Some(cipher) => Ok(cipher.decrypt_value_typed(event.into_serde()?)?),
+        None => Ok(event.into_serde()?),
+    }
+}
+
+async fn v1_2_fix_store(
+    store: &IdbObjectStore<'_>,
+    store_cipher: Option<&StoreCipher>,
+) -> Result<()> {
+    fn maybe_fix_json(raw_json: &RawJsonValue) -> Result<Option<Box<RawJsonValue>>> {
+        let json = raw_json.get();
+
+        if json.contains(r#""content":null"#) {
```

Comment:
I know this is a bit paranoid but `"content":null` could also be part of user-defined data (i.e. nested under a non-null `"content"`), can you add `&& !json.contains(r#""content":{"#)` so we don't do the replacement in that case? 😄

Reply:
Ah right this doesn't work because of the content of `unsigned`... I guess the only option is to deserialize the JSON to fix it?

Reply:
Do you think that would make it very slow? It will probably be fine to go back to the previous solution.

Reply:
We could check first if there's a `"content":null` and only then try to deserialize to see if we need to fix it.

Reply:
Sounds good!